### PR TITLE
Pmd7 upgrade 7.2.0 remoting set3

### PR DESCRIPTION
**Pull request by jborgers**

### CODE REVIEW

File: src/main/resources/category/java/remoting.xml [hunk 1]
```diff
@@ -808,7 +808,7 @@ class Good {
             When this HttpHost is then used for a route and stored socketConfig for, port 80 is added for the host and
             the socketConfig is stored with the wrong key and will not be used.
             It is typically difficult to find out if the config is actually used. Note that
-            http://github.com/jborgers/http-client-monitor helps here.&#13;
```
Comment (stokpop) [inline src/main/resources/category/java/remoting.xml:806]: Might also be good to check and point to micrometer and available apache http 4 and 5 metrics (works with spring boot actuator).

File: src/main/resources/category/java/remoting.xml [hunk 2]
```diff
@@ -281,25 +281,17 @@ return HttpClientBuilder.create() // good, setConnectionManager called, pool con
...
+//MethodDeclaration[
+    .//MethodCall/(TypeExpression|VariableAccess)[
+        pmd-java:typeIs('org.apache.http.impl.client.HttpClientBuilder')
+        or pmd-java:typeIs('org.apache.http.impl.client.HttpClients')
+        or pmd-java:typeIs('org.apache.http.impl.nio.client.HttpAsyncClientBuilder')
+    ]
+][
+    .//MethodCall[@MethodName='setConnectionManager']
+]
+(: trying to get to the right line, unfortunately, the chaining seems to make it not work :)
```
Comment (stokpop) [inline src/main/resources/category/java/remoting.xml:293]: hmm, is this always an issue with chaining? should we create an issue in pmd project for this?

**Comment by stokpop** [comment]
Looks, good. We may need to check Apache HttpClient 4 rules if they are compatible with Apache HttpClient 5 (future improvement). Two remarks, but approved.

**Review by stokpop — APPROVED** [review]

File: src/main/resources/category/java/remoting.xml (hunk 1 above)
Comment (jborgers) [reply src/main/resources/category/java/remoting.xml:806]: okay, good point, added

File: src/main/resources/category/java/remoting.xml (hunk 2 above)
Comment (jborgers) [reply src/main/resources/category/java/remoting.xml:293]: it pannes in designer as well as in the unit tests, so I guess also in IntellJ. Maybe wait until we checked with the newest pmd (7.6?)

**Pull request merged**